File: {{ cookiecutter.repo_name }}/{{ cookiecutter.module_name }}/features/features.py
from pathlib import Path

from loguru import logger
from tqdm import tqdm

from {{ cookiecutter.module_name }}.config import PROCESSED_DATA_DIR


def main(
    input_path: Path = PROCESSED_DATA_DIR / "dataset.csv",
    output_path: Path = PROCESSED_DATA_DIR / "features.csv",
) -> None:
    """Generate features from processed dataset.

    Parameters
    ----------
    input_path : Path
        Path to input dataset.
    output_path : Path
        Path to save generated features.
    """
    logger.info("Generating features from dataset...")
    for i in tqdm(range(10), total=10):
        if i == 5:
            logger.info("Something happened for iteration 5.")
    logger.success("Features generation complete.")


if __name__ == "__main__":
    main()

File: {{ cookiecutter.repo_name }}/src/features/features.jl
"""
Feature engineering - mirrors {{ cookiecutter.module_name }}/features/features.py
"""

using DataFrames
using Logging

include("../config.jl")

@info "Creating features..."

# Example: add features to DataFrame
# df[!, :new_feature] = df.x .* 2

@info "Feature engineering complete."
